File: computer-science/search/linear-search/LinearSearch.py
"""
This is a Linear Search searching algorithm that searches each element of an
arraylinearly to find the target item

  - by Michael Sharp
  - michael@softwareontheshore.com
  - www.softwareontheshore.com
"""

numArray = [ 12, 122, 222, 4, 5, 9, 12323, 99, 10, 1, 78, 123, 64234,
                  6663, 99, 46, 2, 124, 18, 22, 144, 1235, 29772, 1912873,
                  -19, 928282, 4759302, 0, 2938, 21, 222, 98, 1293, 8483,
                  988, -9, 8888, -22, 999999, 22222 ]

def linearSearch(array, target):
    a = array
    length = len(array)

    for x in xrange(0, length):
        element = a[x]

        if element == target: return x

    return -1

print linearSearch(numArray, 9)

File: search/linear-search/LinearSearch.js
/*
  This is a Linear Search searching algorithm that searches each element of an
  array linearly to find the target item
*/

'use strict';

const numArray = [ 12, 122, 222, 4, 5, 9, 12323, 99, 10, 1, 78, 123, 64234,
                  6663, 99, 46, 2, 124, 18, 22, 144, 1235, 29772, 1912873,
                  -19, 928282, 4759302, 0, 2938, 21, 222, 98, 1293, 8483,
                  988, -9, 8888, -22, 999999, 22222 ];

function linearSearch(array, target) {
  const a = array;
  const t = target;
  const l = array.length;

  for (let i = 0; i < l; i++) {
    const element = a[i];

    if (element === t) return i;
  }

  return -1;
}

console.log(linearSearch(numArray, 13));
